jingdai/mailroom.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+
+def send_a_thankyou(donation):
+
+    user_prompt1=input("type List or a name > ")
+
+    if user_prompt1=='List':
+        print(donation)
+        return
+#     If the user types a name not in the list, add that name to the data structure and use it.
+# create a name list
+    names=[]
+    for pair in donation:
+        if pair[0] not in names:
+            names.append(pair[0])
+#print(name)
+
+    if user_prompt1 in names:
+        user_prompt2=input("Amount > ")
+        new_pair=(user_prompt1,float(user_prompt2))
+        donation.append(new_pair)
+    else:
+        new_pair1 = (user_prompt1,0)
+        donation.append(new_pair1)
+        names.append(user_prompt1)
+
+    for name in names:
+        print("Dear {}, thank you for your generous donation.".format(name))
```

Comment:
Requirement stated that we just want to send an email to the person you just added/created the donation for.  Not an email to all the donors.